packages/babel-plugin-transform-flow-strip-types/src/index.js

```diff
@@ -57,7 +57,16 @@ export default function ({ types: t }) {
           node = node.expression;
         } while (t.isTypeCastExpression(node));
         path.replaceWith(node);
-      }
+      },
+
+      ImportSpecifier(path) {
```

Comment:
I was looking into something unrelated today and came across something that make be a more general fix for this. How would you feel about moving this logic into https://github.com/babel/babel/blob/f611cab0f517351e8862581e483ba44a77dcc611/packages/babel-traverse/src/path/lib/virtual-types.js#L107

Then the `Flow(path){ path.remove(); }` above will automatically do this.